### ISSUE TITLE
Fix `custom-property-no-missing-var-function` false positives for recently introduced properties

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#9469]
Ref: https://github.com/stylelint/stylelint/pull/9466

> Two related gaps are out of scope here because they belong to other specs: `scroll-timeline-name` / `view-timeline-name` (and their shorthands)

We often have to update this rule to manually add properties that accept `<dashed-ident>`s to the `Set`.

> The same grammar is in the css-tree syntax data Stylelint already ships,

We could lessen the burden by using csstree's grammar:

```js
IGNORED_PROPERTIES = Object.keys(lexer.properties).filter(p => !lexer.matchProperty(p, '--foo').error)
```

The change/refactor will fix the false positives for things like `scroll-timeline-name`, so we should classifying it as bug fix rather than a refactor.

**Comment by ybiquitous** [comment]
Sounds good. Let's simplify it. 👍🏼
